grpc/observability/config.proto

```diff
@@ -0,0 +1,142 @@
+// Copyright 2022 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Observability Config is used by gRPC Observability plugin to control provided
+// observability features. It contains parameters to enable/disable certain
+// features, or fine tune the verbosity.
+//
+// Note that gRPC may use this config in JSON form, not in protobuf form. This
+// proto definition is intended to help document the schema but might not
+// actually be used directly by gRPC.
+
+syntax = "proto3";
+
+package grpc.observability.config;
+
+import "google/protobuf/duration.proto";
+import "grpc/observability/logging/v1/logging.proto";
+
+option java_package = "io.grpc.observability.config";
+option java_multiple_files = true;
+option java_outer_classname = "ObservabilityCOnfig";
```

Comment:
This should end in "Proto". The convention of outer class name is FileNameProto, which reuses the file namespace to avoid collisions.

```
$ git grep java_outer_classname
grpc/binlog/v1/binarylog.proto:option java_outer_classname = "BinaryLogProto";
grpc/binlog/v1alpha/binarylog.proto:option java_outer_classname = "BinaryLogProto";
grpc/channelz/v1/channelz.proto:option java_outer_classname = "ChannelzProto";
grpc/examples/helloworld.proto:option java_outer_classname = "HelloWorldProto";
grpc/gcp/altscontext.proto:option java_outer_classname = "AltsContextProto";
grpc/gcp/handshaker.proto:option java_outer_classname = "HandshakerProto";
grpc/gcp/transport_security_common.proto:option java_outer_classname = "TransportSecurityCommonProto";
grpc/health/v1/health.proto:option java_outer_classname = "HealthProto";
grpc/lb/v1/load_balancer.proto:option java_outer_classname = "LoadBalancerProto";
grpc/lookup/v1/rls.proto:option java_outer_classname = "RlsProto";
grpc/lookup/v1/rls_config.proto:option java_outer_classname = "RlsConfigProto";
grpc/reflection/v1/reflection.proto:option java_outer_classname = "ServerReflectionProto";
grpc/reflection/v1alpha/reflection.proto:option java_outer_classname = "ServerReflectionProto";
grpc/service_config/service_config.proto:option java_outer_classname = "ServiceConfigProto";
grpc/testing/benchmark_service.proto:option java_outer_classname = "BenchmarkServiceProto";
grpc/testing/control.proto:option java_outer_classname = "ControlProto";
grpc/testing/empty.proto:option java_outer_classname = "EmptyProtos";
grpc/testing/payloads.proto:option java_outer_classname = "PayloadsProto";
grpc/testing/report_qps_scenario_service.proto:option java_outer_classname = "ReportQpsScenarioServiceProto";
grpc/testing/stats.proto:option java_outer_classname = "StatsProto";
grpc/testing/worker_service.proto:option java_outer_classname = "WorkerServiceProto";
grpc/tls/provider/meshca/experimental/config.proto:option java_outer_classname = "MeshCaConfigProto";
```

`MeshCaConfigProto` is the only exception there.

Reply:
Done and done.